engine/pkg/externalresource/manager/gc_coordinator.go

```diff
@@ -175,68 +172,59 @@ func (c *DefaultGCCoordinator) gcByStatusSnapshots(
 	}
 
 	var (
-		toGC []engineModel.JobID
-
-		// toRemove is used to remove meta records when
-		// the associated executors are offline.
-		// TODO adjust the mechanism when we implement S3 support.
-		toRemove []model.ExecutorID
+		toGCJobs      []engineModel.JobID
+		toGCExecutors []model.ExecutorID
 	)
 	for _, resMeta := range resources {
 		if _, exists := jobSnapshot[resMeta.Job]; !exists {
 			// The resource belongs to a deleted job.
-			toGC = append(toGC, resMeta.Job)
+			toGCJobs = append(toGCJobs, resMeta.Job)
 			continue
 		}
 
 		if _, exists := executorSet[resMeta.Executor]; !exists {
 			// The resource belongs to an offlined executor.
-			toRemove = append(toRemove, resMeta.Executor)
+			tp, resName, err := resModel.PasreResourceID(resMeta.ID)
```

Comment:
Since we query resources after getting the executor's snapshot, is it possible that a new executor creates a resource but the executor is not in the `executorSet`?

Reply:
Yes, it is possible, so we need to query meta before getting the snapshot.